### PR TITLE
[WIP] 376 modis work

### PR DESCRIPTION
**Pull request by bhbraswell**

### CODE REVIEW

File: gips/scripts/export.py [hunk 1]
```diff
@@ -30,9 +32,113 @@
...
 from gips.inventory import orm
 
+# from backports import tempfile
+import tempfile
+import boto3
+import zipfile
+import shutil
+
+
+def get_s3_shppath(s3path, tmpdir):
```
Comment (ircwaves) [inline gips/scripts/export.py:39]: This is great.  Maybe we can move it to gips.utils and have `open_vector` use it automagically to make all of gips cli s3 enabled.

**Comment by bhbraswell** [comment]
Added some more commits. This works
`gips_export landsat -p ndvi -d 2019-04-01 -s gips/test/NHseacoast.shp -v4 --outdir /archive/export/test --notld --res 500 500 --fetch`

**Comment by ircwaves** [comment]
Awesome!  Does it produce something `gdalinfo -stats`-similar to what gips 0.14.5-dev produces with the same command line?

**Comment by bhbraswell** [comment]
Yeah, looks OK

```
root@84b0dccb5d6f:/gips# gdalinfo -stats /archive/export/test/0/2019091_LC8_ndvi.tif
Driver: GTiff/GeoTIFF
Files: /archive/export/test/0/2019091_LC8_ndvi.tif
Size is 95, 132
Coordinate System is:
PROJCS["WGS 84 / UTM zone 16N",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",-87],
    PARAMETER["scale_factor",0.9996],
    PARAMETER["false_easting",500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["Easting",EAST],
    AXIS["Northing",NORTH],
    AUTHORITY["EPSG","32616"]]
Origin = (1777848.526153343031183,4934179.330971064977348)
Pixel Size = (497.341116563034689,-496.661877203485801)
Metadata:
  AOD Source=MODIS (MOD08_D3)
  AOD Value=0.18800000893
  AREA_OR_POINT=Point
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  ( 1777848.526, 4934179.331) ( 71d12'50.24"W, 43d27'22.98"N)
Lower Left  ( 1777848.526, 4868619.963) ( 71d21'48.26"W, 42d53'17.27"N)
Upper Right ( 1825095.932, 4934179.331) ( 70d39'11.44"W, 43d22'31.64"N)
Lower Right ( 1825095.932, 4868619.963) ( 70d48'26.53"W, 42d48'31.62"N)
Center      ( 1801472.229, 4901399.647) ( 71d 0'35.07"W, 43d 7'57.27"N)
Band 1 Block=95x43 Type=Int16, ColorInterp=Gray
  Minimum=-14682.000, Maximum=8271.000, Mean=3938.485, StdDev=2308.150
  NoData Value=-32768
  Offset: 0,   Scale:0.0001
  Metadata:
    STATISTICS_MAXIMUM=8271
    STATISTICS_MEAN=3938.4845199508
    STATISTICS_MINIMUM=-14682
    STATISTICS_STDDEV=2308.1496805605
    STATISTICS_VALID_PERCENT=71.35
```

image

**Comment by bhbraswell** [comment]
I'll check against `0.14.5-dev`

**Comment by bhbraswell** [comment]
Old (Python 2.7.12, gippy 0.3.12)
    STATISTICS_MAXIMUM=32767
    STATISTICS_MEAN=3926.5803077389
    STATISTICS_MINIMUM=-32264
    STATISTICS_STDDEV=2302.3938006975

New (Python 3.6.7, gippy 1.0.3)
    STATISTICS_MAXIMUM=8271
    STATISTICS_MEAN=3938.4845199508
    STATISTICS_MINIMUM=-14682
    STATISTICS_STDDEV=2308.1496805605

@ircwaves ^^^^

**Comment by ircwaves** [comment]
Those stats look like `gippy==0.3.11` statistics, before I fixed the chunking bug.  True?

**Comment by bhbraswell** [comment]
I'd be interested to look at those 3 conflicted files with you when you want.

**Comment by ircwaves** [comment]
I love it when stuff is repeatable.

The scene in the archive when I run gets a different AOD value, and it looks like you are missing some of the "metadata snowballing":
```
  AOD Source=MODIS (MOD08_D3)
  AOD Value=0.076
  AREA_OR_POINT=Area
  GIPS_Landsat_Version=1.0.1
  GIPS_Source_Assets=LC08_L1TP_012030_20190401_20190421_01_T1_gs.json
  GIPS_Version=0.14.5-dev
```
The stats of the exported file are different too:
```
    STATISTICS_MAXIMUM=7228
    STATISTICS_MEAN=3704.338954278
    STATISTICS_MINIMUM=-5087
    STATISTICS_STDDEV=1875.8245062027
```
I guess that part will have to be done a little more in depth than `gdalinfo  -stats`.

**Comment by bhbraswell** [comment]
Are you saying that when you run code from this branch you get a different result? We should be able to reconcile that.

**Comment by bhbraswell** [comment]
OK here's the gory details of my test case, using the most recent pushed code.
```
root@ac3e4cb66c08:/gips# find /archive/landsat/ -name "*.aux*" -exec rm {} \;
root@ac3e4cb66c08:/gips# find /archive/landsat/ -name "*.tif*" -exec rm {} \;
root@ac3e4cb66c08:/gips# gips_export landsat -p ndvi -d 2019-04-01 -s gips/test/NHseacoast.shp -v4 --outdir /archive/export/test --notld --res 30 30 --fetch --overwrite
GIPS Data Export (v0.14.5-dev)
Retrieving inventory for site NHseacoast-0 for date range 2019-04-01 - 2019-04-01 (days 1-366)
Attempting to load LC08_L1TP_012030_20190401_20190421_01_T1.tar.gz
Attempting to load LC08_L1TP_012030_20190401_20190421_01_T1.tar.gz
Processing [ndvi] on 1 dates (1 files)
Extracting files from /archive/landsat/tiles/012030/2019091/LC08_L1TP_012030_20190401_20190421_01_T1.tar.gz to /archive/landsat/tiles/012030/2019091/extract7spmf1bi
LC08_L1TP_012030_20190401_20190421_01_T1_MTL.txt exists, not extracting
No files to extract or move
tempdir: infintesimal delay on deletion
0:00:00.000019:  Gathering band files
0:00:00.000383:  reading bands
0:00:26.570829:  done reading bands
LC08_L1TP_012030_20190401_20190421_01_T1_B1: read in 0:00:26.665663
Running atmospheric model (6S)
Retrieving inventory for site tiles for date range 2019-04-01 - 2019-04-01 (days 1-366)
MOD08_D3.A2019091.061.2019092084410[0]: read (0,0)-(0,0) in 0.006568 seconds
AOD: Source = MODIS (MOD08_D3) Value = 0.18800000893
  Band        T      Lu      Ld
COASTAL:    0.999   34.12  146.28
  BLUE:    0.994   27.31  164.82
 GREEN:    0.959   14.38  154.95
   RED:    0.973    8.39  144.10
   NIR:    1.000    2.75  105.70
 SWIR1:    0.985    0.15   27.28
 SWIR2:    0.968    0.02    8.79
CIRRUS:    0.072    0.06    0.69
Ran atmospheric model in 0:00:06.010544
Starting on 1 indices: dict_keys(['ndvi'])
Starting on ndvi
GIPPY: Indices
Colors used:  nir red
Chunk (0,0)-(7881,2128) of 63055881
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,0)-(7881,2128) in 1.70141 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B5[0]: read (0,0)-(7881,2128) in 2.89761 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,0)-(7881,2128) in 0.146229 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B4[0]: read (0,0)-(7881,2128) in 1.01118 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,0)-(7881,2128) in 0.144074 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,0)-(7881,2128) in 0.141247 seconds
012030_2019091_LC8_ndvi[0]: Writing (0.0001x + 0)
Chunk (0,2128)-(7881,4256) of 63055881
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,2128)-(7881,4256) in 1.67777 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B5[0]: read (0,2128)-(7881,4256) in 2.77979 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,2128)-(7881,4256) in 0.134091 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B4[0]: read (0,2128)-(7881,4256) in 0.986123 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,2128)-(7881,4256) in 0.143904 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,2128)-(7881,4256) in 0.140589 seconds
Chunk (0,4256)-(7881,6384) of 63055881
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,4256)-(7881,6384) in 1.41142 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B5[0]: read (0,4256)-(7881,6384) in 2.48319 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,4256)-(7881,6384) in 0.152036 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B4[0]: read (0,4256)-(7881,6384) in 0.878987 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,4256)-(7881,6384) in 0.135488 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,4256)-(7881,6384) in 0.145708 seconds
Chunk (0,6384)-(7881,8001) of 63055881
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,6384)-(7881,8001) in 0.775268 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B5[0]: read (0,6384)-(7881,8001) in 1.33384 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,6384)-(7881,8001) in 0.102524 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_B4[0]: read (0,6384)-(7881,8001) in 0.798672 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,6384)-(7881,8001) in 0.103097 seconds
LC08_L1TP_012030_20190401_20190421_01_T1_BQA[0]: read (0,6384)-(7881,8001) in 0.10679 seconds
 -> 012030_2019091_LC8: processed dict_keys(['ndvi']) in 0:00:17.735252
tempdir: infintesimal delay on deletion
Processing completed in 0:00:51.881135
Creating mosaic project /archive/export/test/0
  Dates: 1 dates (2019-04-01 - 2019-04-01)
  Products: ndvi
GIPPY: cookie_cutter (1 files) - /archive/export/test/0/mosaic1yqxl34g/2019091_LC8_ndvi.tif
012030_2019091_LC8_ndvi warping into 2019091_LC8_ndvi
012030_2019091_LC8_ndvi[0] warping into 2019091_LC8_ndvi[0]
0...10...20...30...40...50...60...70...80...90...100 - done.
tempdir: infintesimal delay on deletion
2019-04-01: created project files for 1 tiles in 0:00:03.006392
Completed mosaic project in 0:00:03.008619
/archive/export/test/0/2019091_LC8_ndvi.tif
/archive/export/test/0/2018101_MCD_ndvi.tif
/archive/export/test/0/2019091_LC8_ndvi.tif.aux.xml
    DATE     Coverage  Product  
2018        
    101     ndvi  
2019        
    091     ndvi  ndvi.tif.aux  


3 files on 2 dates
root@ac3e4cb66c08:/gips# gdalinfo -stats /archive/export/test/0/2019091_LC8_ndvi.tif
Driver: GTiff/GeoTIFF
Files: /archive/export/test/0/2019091_LC8_ndvi.tif
       /archive/export/test/0/2019091_LC8_ndvi.tif.aux.xml
Size is 1575, 2186
Coordinate System is:
PROJCS["WGS 84 / UTM zone 16N",
    GEOGCS["WGS 84",
        DATUM["WGS_1984",
            SPHEROID["WGS 84",6378137,298.257223563,
                AUTHORITY["EPSG","7030"]],
            AUTHORITY["EPSG","6326"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4326"]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",-87],
    PARAMETER["scale_factor",0.9996],
    PARAMETER["false_easting",500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AXIS["Easting",EAST],
    AXIS["Northing",NORTH],
    AUTHORITY["EPSG","32616"]]
Origin = (1777848.526153343031183,4934179.330971064977348)
Pixel Size = (29.998353062532249,-29.990561660960715)
Metadata:
  AOD Source=MODIS (MOD08_D3)
  AOD Value=0.18800000893
  AREA_OR_POINT=Point
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  ( 1777848.526, 4934179.331) ( 71d12'50.24"W, 43d27'22.98"N)
Lower Left  ( 1777848.526, 4868619.963) ( 71d21'48.26"W, 42d53'17.27"N)
Upper Right ( 1825095.932, 4934179.331) ( 70d39'11.44"W, 43d22'31.64"N)
Lower Right ( 1825095.932, 4868619.963) ( 70d48'26.53"W, 42d48'31.62"N)
Center      ( 1801472.229, 4901399.647) ( 71d 0'35.07"W, 43d 7'57.27"N)
Band 1 Block=1575x2 Type=Int16, ColorInterp=Gray
  Min=-14682.000 Max=8271.000 
  Minimum=-14682.000, Maximum=8271.000, Mean=3938.485, StdDev=2308.150
  NoData Value=-32768
  Offset: 0,   Scale:0.0001
  Metadata:
    STATISTICS_MAXIMUM=8271
    STATISTICS_MEAN=3938.4845199508
    STATISTICS_MINIMUM=-14682
    STATISTICS_STDDEV=2308.1496805605
    STATISTICS_VALID_PERCENT=71.35
```

**Comment by ircwaves** [comment]
Not saying that.  I was noting that my `0.14.5-dev` results differ in statistics, metadata values (AOD), and number of metadata entries (mine 6, yours 3).  I'll roll a container using your dockerfile to make a more reproducible comparison...my venv is at least 15 months old, and who knows what state my OS is in ;)

**Comment by bhbraswell** [comment]
Getting a lot of these
```
ERROR 6: /vsitar//archive/landsat/tiles/012030/2018200/LC08_L1TP_012030_20180719_20180731_01_T1.tar.gz/LC08_L1TP_012030_20180719_20180731_01_T1_B1.TIF, band 1: SetOffset() not supported on this raster band.
ERROR 6: /vsitar//archive/landsat/tiles/012030/2018200/LC08_L1TP_012030_20180719_20180731_01_T1.tar.gz/LC08_L1TP_012030_20180719_20180731_01_T1_B1.TIF, band 1: SetScale() not supported on this raster band.
ERROR 6: /vsitar//archive/landsat/tiles/012030/2018200/LC08_L1TP_012030_20180719_20180731_01_T1.tar.gz/LC08_L1TP_012030_20180719_20180731_01_T1_B1.TIF, band 1: SetUnitType() not supported on this raster band.
```

**Comment by ircwaves** [comment]
Those have been happening all along.  They come from reading the rasterband right out of the tar file, and then adding a scale and offset to the GeoRaster object.    The band is essentially a read-only band, but that aspect is handled poorly here by saying "`ERROR 6: .... [ you can't do that to a band in a tar file ]`".

**Comment by bhbraswell** [comment]
I'm going to pause for a while. I think I'm beginning to see diminishing returns, and also too long awol from my actual assigned tasks. If possible it would be great to talk about next steps at some point soon.

**Pull request closed without merging**